web/controllers/friends_controller.ex

```diff
@@ -5,34 +5,54 @@ defmodule Entice.Web.FriendsController do
 
   plug :ensure_login
 
+  #Case to think about:
+  #Player A adds friend Player B under name N
+  #Player B deletes account (and so all chars are deleted)
+  #Player C creates acccount and char under name N
+  #Player A adds Player C under name N
+  #Now in friends DB: (N, PlayerA.id, PlayerB.id) and (N, PlayerA.id, PlayerC.id)
+  #Since all queries are done by name it'll be an issue.
+  #Solutions:
+  #Delete all friends with PlayerB.id in them
+
+
+  #TODO: add map once it's server sided, order by creation date
   @doc "Returns all friends of connected account."
   def index(conn, _params) do
     id = get_session(conn, :client_id)
     {:ok, friends} = Entice.Web.Client.get_friends(id)
 
-    friends = for friend <- friends, do: friend.friend_account.id
+    results = []
+    results = for friend <- friends do
+      {:ok, status, name} = get_status(friend.base_name)
+      map = %{base_name: friend.base_name, current_name: name, status: status}
+      results = results ++ [map]
+    end
+
     conn |> json ok(%{
       message: "All friends",
-      friends: friends})
+      friends: results})
   end
 
+  defp get_status(friend_name) do
+    {:ok, status, name} = Client.get_status(friend_name)
+  end
 
   @doc "Adds friend :id to friends list of connected account."
   def create(conn, _params) do
     session_id = get_session(conn, :client_id)
     {:ok, acc} = Client.get_account(session_id)
     account_id = acc.id
-    friend_account_id = conn.params["friend_account_id"]
-
-    friend_account = Entice.Web.Repo.get(Entice.Web.Account, friend_account_id)
+    friend_name = conn.params["char_name"]
```

Comment:
we can match on this in the function header. replace `_params` with a `%{"char_name" => friend_name}`

and dont forget to add catch all def for the create function below, so we can handle wrong parameters gracefully. its implemented like this in the other controllers already, check account controller for example